### DISCUSSION TITLE
rate limiting, prevent DDOS, malicious users?

### DISCUSSION THREAD

**Discussion by dragonballa** [nats-io/nats-server#5425] [General]
I want to know if its possible to rate limit publishers or consumers from overloading/DDOS/malicious behavior without having to build my own gateways.

I found somebody exactly in my shoes here: https://github.com/nats-io/nats-server/issues/5092 

Can we use KV to throttle users ? I'm not sure what the solution here is without using gateways like `node-rate-limiter-flexible`

**Comment by derekcollison** [comment]
Nothing exists today as I mentioned in the issue you also posted to.
